lib/src/chart/line_chart/line_chart_data.dart

```diff
@@ -198,4 +204,62 @@ class FlDotData {
     this.dotSize = 4.0,
     this.checkToShowDot = showAllDots,
   });
-}
\ No newline at end of file
+}
+
+class DashDefinition {
+  final double solidWidth;
+  final double gapWidth;
+
+  const DashDefinition({this.solidWidth, this.gapWidth});
+}
+
+class LineChartLine extends FlLine {
+  final DashDefinition dashDefinition;
+
+  const LineChartLine({
+    Color color,
+    double strokeWidth,
+    this.dashDefinition,
+  }) : super(color: color, strokeWidth: strokeWidth);
+}
+
+class VerticalLine extends LineChartLine {
+  final double x;
+  final double endY;
+
+  const VerticalLine({
+    @required this.x,
+    this.endY,
+    DashDefinition dashDefinition,
+    Color color = Colors.black,
+    double strokeWidth = 1,
+  }) : super(color: color, strokeWidth: strokeWidth, dashDefinition: dashDefinition);
+}
+
+class HorizontalLine extends LineChartLine {
+  final double y;
+  final double endX;
```

Comment:
also we don't need endX, (read previous comment)

Reply:
What if the user wants the line to go all the way to the top though and not necessarily end at the data point?

Reply:
Drop it plz, if we want to make it possible to set end point then we have to add start point for it to be customizable

Reply:
Dropped